state/remote/state.go

```diff
@@ -123,8 +138,12 @@ func (s *State) PersistState() error {
 	defer s.mu.Unlock()
 
 	if s.readState != nil {
-		if statefile.StatesMarshalEqual(s.state, s.readState) {
-			// If the state hasn't changed at all then we have nothing to do.
+		lineageUnchanged := s.readLineage != "" && s.lineage == s.readLineage
+		serialUnchanged := s.readSerial != 0 && s.serial == s.readSerial
+		stateUnchanged := statefile.StatesMarshalEqual(s.state, s.readState)
+		if stateUnchanged && lineageUnchanged && serialUnchanged {
```

Comment:
I appreciate the addition of these more granular checks -- we'll probably want to note this in the changelog as part of the changes included in this changeset (more specific checks for whether state needs to be updated)

Reply:
👍 will do